spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -59,12 +67,49 @@ public DefaultSubscriberFactory(GcpProjectIdProvider projectIdProvider,
 	}
 
 	@Override
-	public Subscriber getSubscriber(String subscriptionName, MessageReceiver receiver) {
+	public Subscriber createSubscriber(String subscriptionName, MessageReceiver receiver) {
 		return Subscriber.newBuilder(
 				SubscriptionName.of(this.projectId, subscriptionName), receiver)
 				.setChannelProvider(this.channelProvider)
 				.setExecutorProvider(this.executorProvider)
 				.setCredentialsProvider(this.credentialsProvider)
 				.build();
 	}
+
+	@Override
+	public PullRequest createPullRequest(String subscriptionName, Integer maxMessages,
+			Boolean returnImmediately) {
+		PullRequest.Builder pullRequestBuilder =
+				PullRequest.newBuilder().setSubscriptionWithSubscriptionName(
+						SubscriptionName.of(this.projectId, subscriptionName));
+
+		if (maxMessages != null) {
+			pullRequestBuilder.setMaxMessages(maxMessages);
+		}
+
+		if (returnImmediately != null) {
+			pullRequestBuilder.setReturnImmediately(returnImmediately);
+		}
+
+		return pullRequestBuilder.build();
+	}
+
+	@Override
+	public SubscriberStub createSubscriberStub(RetrySettings retrySettings) {
+		try {
+			SubscriptionAdminSettings.Builder subscriptionAdminSettingsBuilder = SubscriptionAdminSettings.newBuilder()
+					.setExecutorProvider(this.executorProvider)
+					.setCredentialsProvider(this.credentialsProvider)
+					.setTransportChannelProvider(this.channelProvider);
+
+			if (retrySettings != null) {
+				subscriptionAdminSettingsBuilder.pullSettings().setRetrySettings(retrySettings);
+			}
+
+			return GrpcSubscriberStub.create(subscriptionAdminSettingsBuilder.build());
+		}
+		catch (IOException e) {
+			throw new RuntimeException("Error creating default SubscriptionAdminSettings", e);
```

Comment:
This message for exception is misleading somehow. We expect `SubscriberStub` as an output but the message says something about some unrelated `SubscriptionAdminSettings`. Moreover *default*... :shrugged: